File: dataset/human/java/file_332.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        int[] nums = new int[n];

        int numNeg = 0;
        long sum = 0;
        int minMag = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();

            sum += Math.abs(nums[i]);

            if (nums[i] < 0) {
                numNeg++;
            }

            minMag = Math.min(minMag, Math.abs(nums[i]));
        }

        if (numNeg % 2 == 0) {
            System.out.println(sum);
        } else {
            System.out.println(sum - 2 * minMag);
        }
    }
}

File: dataset/human/python/file_332.py
import numpy as np
N = int(input())
list1 = list(map(int, input().split()))

sum1 = 0
for i in list1:
    if(i<0):
        sum1 += 1

nl1 = np.array(list1)
np.abs(nl1)
if(sum1%2 == 0):
    a = sum(np.abs(nl1))
else:
    a = sum(np.abs(nl1))-2*min(np.abs(nl1))

print(a)
